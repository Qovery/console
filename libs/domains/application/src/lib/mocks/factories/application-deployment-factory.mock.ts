import { DeploymentHistoryApplication, StateEnum } from 'qovery-typescript-axios'
import { Chance } from 'chance'

const chance = new Chance()

export const applicationDeploymentsFactoryMock = (howMany: number): DeploymentHistoryApplication[] =>
  Array.from({ length: howMany }).map((_, index) => ({
    id: index.toString(),
    created_at: new Date().toString(),
    updated_at: new Date().toString(),
    name: chance.name(),
    status: chance.pickone(Object.values([StateEnum.DEPLOYED, StateEnum.RUNNING, StateEnum.STOP_ERROR])),
  }))
