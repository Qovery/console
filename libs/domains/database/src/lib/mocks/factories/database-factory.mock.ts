import { Chance } from 'chance'
import { Database, DatabaseTypeEnum, DatabaseModeEnum, DatabaseAccessibilityEnum } from 'qovery-typescript-axios'

const chance = new Chance()

export const databaseFactoryMock = (howMany: number): Database[] =>
  Array.from({ length: howMany }).map((_, index) => ({
    id: `${index}`,
    created_at: new Date().toString(),
    updated_at: new Date().toString(),
    name: chance.name(),
    type: DatabaseTypeEnum.POSTGRESQL,
    version: '12',
    mode: DatabaseModeEnum.CONTAINER,
    accessibility: DatabaseAccessibilityEnum.PRIVATE,
    cpu: 1000,
    memory: 1024,
    storage: 1024,
    environment: {
      id: chance.guid(),
    },
    host: chance.name(),
    port: 80,
    maximum_cpu: 10,
    maximum_memory: 10,
    disk_encrypted: false,
  }))
