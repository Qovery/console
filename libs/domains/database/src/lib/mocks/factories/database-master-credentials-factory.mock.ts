import { Chance } from 'chance'
import { Credentials } from 'qovery-typescript-axios'

const chance = new Chance()

export const databaseMasterCredentialsFactoryMock = (howMany: number): Credentials[] =>
  Array.from({ length: howMany }).map((_, index) => ({
    login: chance.name(),
    password: chance.guid(),
    host: chance.name(),
    port: chance.d4(),
  }))
