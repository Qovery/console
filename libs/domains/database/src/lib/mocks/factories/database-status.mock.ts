import { Chance } from 'chance'
import { Status, ServiceDeploymentStatusEnum, StateEnum } from 'qovery-typescript-axios'

const chance = new Chance()

export const databaseStatusFactoryMock = (howMany: number): Status[] =>
  Array.from({ length: howMany }).map((_, index) => ({
    id: `${index}`,
    message: chance.sentence(),
    service_deployment_status: chance.pickone(
      Object.values([
        ServiceDeploymentStatusEnum.UP_TO_DATE,
        ServiceDeploymentStatusEnum.UP_TO_DATE,
        ServiceDeploymentStatusEnum.UP_TO_DATE,
      ])
    ),
    state: chance.pickone(Object.values([StateEnum.DEPLOYED, StateEnum.RUNNING, StateEnum.STOP_ERROR])),
  }))
