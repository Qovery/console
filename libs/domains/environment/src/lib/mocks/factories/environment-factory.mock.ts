import { Chance } from 'chance'
import {
  DeploymentHistoryStatusEnum,
  EnvironmentModeEnum,
  ServiceDeploymentStatusEnum,
  StateEnum,
  WeekdayEnum,
} from 'qovery-typescript-axios'
import { EnvironmentEntity } from '@qovery/shared/interfaces'

const chance = new Chance()

const service = {
  id: chance.integer().toString(),
  created_at: chance.date().toString(),
  updated_at: chance.date().toString(),
  name: chance.name(),
  status: chance.pickone(Object.values([StateEnum.DEPLOYED, StateEnum.RUNNING, StateEnum.STOP_ERROR])),
}

const database = {
  id: chance.integer().toString(),
  created_at: chance.date().toString(),
  updated_at: chance.date().toString(),
  name: chance.name(),
  status: chance.pickone(Object.values([DeploymentHistoryStatusEnum.SUCCESS, DeploymentHistoryStatusEnum.FAILED])),
}

const deployment = {
  id: chance.integer().toString(),
  created_at: chance.date().toString(),
  updated_at: chance.date().toString(),
  status: StateEnum.DEPLOYED,
  applications: [service, service, service],
  databases: [database, database, database],
}

const deploymentRules = {
  id: chance.guid(),
  created_at: chance.date().toString(),
  updated_at: chance.date().toString(),
  auto_deploy: true,
  auto_stop: true,
  auto_delete: true,
  auto_preview: true,
  timezone: 'UTC',
  start_time: chance.date().toString(),
  stop_time: chance.date().toString(),
  weekdays: [chance.pickone(Object.values(WeekdayEnum))],
}

export const environmentFactoryMock = (howMany: number, noStatus = false, noDeployments = false): EnvironmentEntity[] =>
  Array.from({ length: howMany }).map((_, index) => ({
    id: `${(index + 1) * Math.random()}`,
    created_at: chance.date().toString(),
    updated_at: chance.date().toString(),
    name: chance.name(),
    last_updated_by: chance.date().toString(),
    cloud_provider: {
      provider: chance.name(),
      cluster: chance.name(),
    },
    mode: chance.pickone(
      Object.values([EnvironmentModeEnum.DEVELOPMENT, EnvironmentModeEnum.PRODUCTION, EnvironmentModeEnum.STAGING])
    ),
    cluster_id: chance.guid(),
    status: !noStatus
      ? {
          id: `${index}`,
          state: chance.pickone(Object.values([StateEnum.DEPLOYED, StateEnum.RUNNING, StateEnum.STOP_ERROR])),
          message: chance.word({ length: 10 }),
          service_deployment_status: chance.pickone(Object.values([ServiceDeploymentStatusEnum.UP_TO_DATE])),
        }
      : undefined,
    deployments: !noDeployments ? [deployment, deployment, deployment] : undefined,
    deploymentRules: deploymentRules,
  }))
