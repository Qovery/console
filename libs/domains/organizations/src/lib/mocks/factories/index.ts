export * from './organizations-factory.mock'
