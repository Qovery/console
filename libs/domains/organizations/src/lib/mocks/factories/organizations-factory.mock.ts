import { OrganizationInterface } from '../../interfaces/organizations.interface'
import { OrganizationPlanType } from '../../enums/organizations.enum'
import { Chance } from 'chance'

const chance = new Chance()

export const organizationsFactoryMock = (howMany: number): OrganizationInterface[] =>
  Array.from({ length: howMany }).map((_, index) => ({
    id: `${index}`,
    created_at: new Date(),
    updated_at: new Date(),
    name: chance.name(),
    description: chance.word({ length: 10 }),
    plan: chance.pickone(Object.values(OrganizationPlanType)),
    website_url: chance.url(),
    repository: chance.name(),
    logo_url: chance.url(),
    icon_url: chance.url(),
    owner: chance.name(),
  }))
