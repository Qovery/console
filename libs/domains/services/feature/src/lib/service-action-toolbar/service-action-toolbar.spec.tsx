import { environmentFactoryMock, helmFactoryMock } from '@qovery/shared/factories'
import { renderWithProviders, screen } from '@qovery/shared/util-tests'
import { ServiceActionToolbar } from './service-action-toolbar'

const mockHelm = helmFactoryMock(1)[0]
const mockEnvironment = environmentFactoryMock(1)[0]

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({ organizationId: '123', projectId: '456', environmentId: '789' }),
  useNavigate: () => jest.fn(),
}))

jest.mock('../hooks/use-service/use-service', () => ({
  useService: () => ({
    data: mockHelm,
  }),
}))

jest.mock('../hooks/use-deployment-status/use-deployment-status', () => {
  return {
    ...jest.requireActual('../hooks/use-deployment-status/use-deployment-status'),
    useDeploymentStatus: () => ({
      data: {
        state: 'READY',
      },
    }),
  }
})

describe('ServiceActionToolbar', () => {
  it('should match manage deployment snapshot', async () => {
    const { userEvent, baseElement } = renderWithProviders(
      <ServiceActionToolbar serviceId={mockHelm.id} environment={mockEnvironment} />,
      {
        container: document.body,
      }
    )
    const buttonManageDeployment = screen.getByLabelText(/manage deployment/i)
    await userEvent.click(buttonManageDeployment)

    expect(baseElement).toMatchSnapshot()
  })

  it('should match other actions snapshot', async () => {
    const { userEvent, baseElement } = renderWithProviders(
      <ServiceActionToolbar serviceId={mockHelm.id} environment={mockEnvironment} />,
      {
        container: document.body,
      }
    )
    const buttonOtherActions = screen.getByLabelText(/other actions/i)
    await userEvent.click(buttonOtherActions)

    expect(baseElement).toMatchSnapshot()
  })
})
