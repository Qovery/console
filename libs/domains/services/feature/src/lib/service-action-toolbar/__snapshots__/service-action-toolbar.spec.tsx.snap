// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceActionToolbar should match manage deployment snapshot 1`] = `
<body
  data-scroll-locked="1"
  style="pointer-events: none;"
>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    aria-orientation="horizontal"
    data-aria-hidden="true"
    data-orientation="horizontal"
    dir="ltr"
    role="toolbar"
    style="outline: none;"
    tabindex="0"
  >
    <button
      aria-controls="radix-:r7:"
      aria-expanded="true"
      aria-haspopup="menu"
      aria-label="Manage Deployment"
      class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 shadow-sm disabled:shadow-none active:shadow-none bg-white dark:bg-neutral-600 outline-yellow-600 h-9 px-3 first:rounded-l last:rounded-r first:border-x border border-yellow-600 text-yellow-600 data-[state=open]:text-yellow-600 text-sm outline-0 min-w-[36px]"
      data-orientation="horizontal"
      data-radix-collection-item=""
      data-state="open"
      id="radix-:r6:"
      tabindex="-1"
      type="button"
    >
      <div
        class="flex h-full w-full items-center justify-center"
        data-state="closed"
      >
        <i
          aria-hidden="true"
          class="fa-solid fa-play mr-4"
        />
        <i
          aria-hidden="true"
          class="fa-solid fa-chevron-down "
        />
      </div>
    </button>
    <a
      class="cursor-pointer duration-100 gap-1 inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 shadow-sm disabled:shadow-none active:shadow-none bg-white dark:bg-neutral-600 outline-brand-500 h-9 px-3 border border-neutral-250 dark:border-neutral-400 dark:text-white dark:hover:[&:not(:active):not(:disabled)]:border-neutral-350 active:bg-neutral-150 disabled:text-neutral-300 disabled:bg-neutral-100 disabled:border-none first:rounded-l last:rounded-r first:border-x border-l-0 hover:[&:not(:active)]:border-neutral-250 text-neutral-350 hover:text-neutral-400 data-[state=open]:text-neutral-400 text-sm outline-0 min-w-[36px]"
      data-orientation="horizontal"
      data-radix-collection-item=""
      data-state="closed"
      href="/organization/123/project/123/environment/3840068565336064/logs/0/service-logs"
      tabindex="-1"
      type="button"
    >
      <i
        aria-hidden="true"
        class="fa-solid fa-scroll "
      />
    </a>
    <button
      aria-expanded="false"
      aria-haspopup="menu"
      aria-label="Other actions"
      class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 shadow-sm disabled:shadow-none active:shadow-none bg-white dark:bg-neutral-600 outline-brand-500 h-9 px-3 border border-neutral-250 dark:border-neutral-400 dark:text-white dark:hover:[&:not(:active):not(:disabled)]:border-neutral-350 active:bg-neutral-150 disabled:text-neutral-300 disabled:bg-neutral-100 disabled:border-none first:rounded-l last:rounded-r first:border-x border-l-0 hover:[&:not(:active)]:border-neutral-250 text-neutral-350 hover:text-neutral-400 data-[state=open]:text-neutral-400 text-sm outline-0 min-w-[36px]"
      data-orientation="horizontal"
      data-radix-collection-item=""
      data-state="closed"
      id="radix-:rc:"
      tabindex="-1"
      type="button"
    >
      <div
        class="flex h-full w-full items-center justify-center"
        data-state="closed"
      >
        <i
          aria-hidden="true"
          class="fa-solid fa-ellipsis-v "
        />
      </div>
    </button>
  </div>
  <div
    data-radix-popper-content-wrapper=""
    dir="ltr"
    style="position: fixed; left: 0px; top: 0px; transform: translate(0px, 8px); min-width: max-content; --radix-popper-available-width: 0px; --radix-popper-available-height: -8px; --radix-popper-anchor-width: 0px; --radix-popper-anchor-height: 0px; --radix-popper-transform-origin: 0% 0px;"
  >
    <div
      aria-labelledby="radix-:r6:"
      aria-orientation="vertical"
      class="flex w-[258px] flex-col gap-1 rounded-md bg-neutral-50 p-2 shadow-[0_0_32px_rgba(0,0,0,0.08)] data-[state=open]:data-[side=bottom]:animate-slidein-up-md-faded data-[state=open]:data-[side=left]:animate-slidein-right-sm-faded data-[state=open]:data-[side=right]:animate-slidein-left-md-faded data-[state=open]:data-[side=top]:animate-slidein-down-md-faded dark:border dark:border-neutral-500 dark:bg-neutral-550"
      data-align="start"
      data-orientation="vertical"
      data-radix-menu-content=""
      data-side="bottom"
      data-state="open"
      dir="ltr"
      id="radix-:r7:"
      role="menu"
      style="outline: none; --radix-dropdown-menu-content-transform-origin: var(--radix-popper-transform-origin); --radix-dropdown-menu-content-available-width: var(--radix-popper-available-width); --radix-dropdown-menu-content-available-height: var(--radix-popper-available-height); --radix-dropdown-menu-trigger-width: var(--radix-popper-anchor-width); --radix-dropdown-menu-trigger-height: var(--radix-popper-anchor-height); pointer-events: auto;"
      tabindex="-1"
    >
      <div
        class="px-3 flex items-center h-8 text-sm font-medium rounded-sm outline-none data-[highlighted]:bg-neutral-150 cursor-pointer hover:bg-neutral-150 text-neutral-400 relative"
        data-orientation="vertical"
        data-radix-collection-item=""
        role="menuitem"
        tabindex="-1"
      >
        <i
          aria-hidden="true"
          class="fa-solid fa-play text-sm mr-3 text-yellow-600"
        />
        Deploy
        <div
          class="absolute right-2"
          data-state="closed"
        >
          <i
            aria-hidden="true"
            class="fa-regular fa-circle-exclamation "
          />
        </div>
      </div>
      <div
        aria-orientation="horizontal"
        class="my-1 h-[1px] w-full bg-neutral-200 dark:bg-neutral-600"
        role="separator"
      />
      <div
        class="px-3 flex items-center h-8 text-sm font-medium rounded-sm outline-none data-[highlighted]:bg-brand-50 data-[highlighted]:text-brand-500 dark:data-[highlighted]:bg-neutral-400 dark:text-neutral-100 cursor-pointer text-neutral-400 hover:bg-brand-50 hover:text-brand-500 dark:hover:bg-neutral-550"
        data-orientation="vertical"
        data-radix-collection-item=""
        role="menuitem"
        tabindex="-1"
      >
        <i
          aria-hidden="true"
          class="fa-solid fa-clock-rotate-left text-sm mr-3 text-brand-500"
        />
        Deploy another chart version
      </div>
    </div>
  </div>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;

exports[`ServiceActionToolbar should match other actions snapshot 1`] = `
<body
  data-scroll-locked="1"
  style="pointer-events: none;"
>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    aria-orientation="horizontal"
    data-aria-hidden="true"
    data-orientation="horizontal"
    dir="ltr"
    role="toolbar"
    style="outline: none;"
    tabindex="0"
  >
    <button
      aria-expanded="false"
      aria-haspopup="menu"
      aria-label="Manage Deployment"
      class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 shadow-sm disabled:shadow-none active:shadow-none bg-white dark:bg-neutral-600 outline-yellow-600 h-9 px-3 first:rounded-l last:rounded-r first:border-x border border-yellow-600 text-yellow-600 data-[state=open]:text-yellow-600 text-sm outline-0 min-w-[36px]"
      data-orientation="horizontal"
      data-radix-collection-item=""
      data-state="closed"
      id="radix-:rp:"
      tabindex="-1"
      type="button"
    >
      <div
        class="flex h-full w-full items-center justify-center"
        data-state="closed"
      >
        <i
          aria-hidden="true"
          class="fa-solid fa-play mr-4"
        />
        <i
          aria-hidden="true"
          class="fa-solid fa-chevron-down "
        />
      </div>
    </button>
    <a
      class="cursor-pointer duration-100 gap-1 inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 shadow-sm disabled:shadow-none active:shadow-none bg-white dark:bg-neutral-600 outline-brand-500 h-9 px-3 border border-neutral-250 dark:border-neutral-400 dark:text-white dark:hover:[&:not(:active):not(:disabled)]:border-neutral-350 active:bg-neutral-150 disabled:text-neutral-300 disabled:bg-neutral-100 disabled:border-none first:rounded-l last:rounded-r first:border-x border-l-0 hover:[&:not(:active)]:border-neutral-250 text-neutral-350 hover:text-neutral-400 data-[state=open]:text-neutral-400 text-sm outline-0 min-w-[36px]"
      data-orientation="horizontal"
      data-radix-collection-item=""
      data-state="closed"
      href="/organization/123/project/123/environment/3840068565336064/logs/0/service-logs"
      tabindex="-1"
      type="button"
    >
      <i
        aria-hidden="true"
        class="fa-solid fa-scroll "
      />
    </a>
    <button
      aria-controls="radix-:r10:"
      aria-expanded="true"
      aria-haspopup="menu"
      aria-label="Other actions"
      class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 shadow-sm disabled:shadow-none active:shadow-none bg-white dark:bg-neutral-600 outline-brand-500 h-9 px-3 border border-neutral-250 dark:border-neutral-400 dark:text-white dark:hover:[&:not(:active):not(:disabled)]:border-neutral-350 active:bg-neutral-150 disabled:text-neutral-300 disabled:bg-neutral-100 disabled:border-none first:rounded-l last:rounded-r first:border-x border-l-0 hover:[&:not(:active)]:border-neutral-250 text-neutral-350 hover:text-neutral-400 data-[state=open]:text-neutral-400 text-sm outline-0 min-w-[36px]"
      data-orientation="horizontal"
      data-radix-collection-item=""
      data-state="open"
      id="radix-:rv:"
      tabindex="-1"
      type="button"
    >
      <div
        class="flex h-full w-full items-center justify-center"
        data-state="closed"
      >
        <i
          aria-hidden="true"
          class="fa-solid fa-ellipsis-v "
        />
      </div>
    </button>
  </div>
  <div
    data-radix-popper-content-wrapper=""
    dir="ltr"
    style="position: fixed; left: 0px; top: 0px; transform: translate(0px, 8px); min-width: max-content; --radix-popper-available-width: 0px; --radix-popper-available-height: -8px; --radix-popper-anchor-width: 0px; --radix-popper-anchor-height: 0px; --radix-popper-transform-origin: 0% 0px;"
  >
    <div
      aria-labelledby="radix-:rv:"
      aria-orientation="vertical"
      class="flex w-[258px] flex-col gap-1 rounded-md bg-neutral-50 p-2 shadow-[0_0_32px_rgba(0,0,0,0.08)] data-[state=open]:data-[side=bottom]:animate-slidein-up-md-faded data-[state=open]:data-[side=left]:animate-slidein-right-sm-faded data-[state=open]:data-[side=right]:animate-slidein-left-md-faded data-[state=open]:data-[side=top]:animate-slidein-down-md-faded dark:border dark:border-neutral-500 dark:bg-neutral-550"
      data-align="start"
      data-orientation="vertical"
      data-radix-menu-content=""
      data-side="bottom"
      data-state="open"
      dir="ltr"
      id="radix-:r10:"
      role="menu"
      style="outline: none; --radix-dropdown-menu-content-transform-origin: var(--radix-popper-transform-origin); --radix-dropdown-menu-content-available-width: var(--radix-popper-available-width); --radix-dropdown-menu-content-available-height: var(--radix-popper-available-height); --radix-dropdown-menu-trigger-width: var(--radix-popper-anchor-width); --radix-dropdown-menu-trigger-height: var(--radix-popper-anchor-height); pointer-events: auto;"
      tabindex="-1"
    >
      <a
        class="transition duration-100 px-3 flex items-center h-8 text-sm font-medium rounded-sm outline-none data-[highlighted]:bg-brand-50 data-[highlighted]:text-brand-500 dark:data-[highlighted]:bg-neutral-400 dark:text-neutral-100 cursor-pointer text-neutral-400 hover:bg-brand-50 hover:text-brand-500 dark:hover:bg-neutral-550 gap-0"
        data-orientation="vertical"
        data-radix-collection-item=""
        href="/organization/123/project/123/environment/3840068565336064/logs/0/service-logs"
        role="menuitem"
        tabindex="-1"
      >
        <i
          aria-hidden="true"
          class="fa-solid fa-scroll text-sm mr-3 text-brand-500"
        />
        Logs
      </a>
      <a
        href="https://gitpod.io/#/tree/"
        rel="noreferrer"
        target="_blank"
      >
        <div
          class="px-3 flex items-center h-8 text-sm font-medium rounded-sm outline-none data-[highlighted]:bg-brand-50 data-[highlighted]:text-brand-500 dark:data-[highlighted]:bg-neutral-400 dark:text-neutral-100 cursor-pointer text-neutral-400 hover:bg-brand-50 hover:text-brand-500 dark:hover:bg-neutral-550"
          data-orientation="vertical"
          data-radix-collection-item=""
          role="menuitem"
          tabindex="-1"
        >
          <i
            aria-hidden="true"
            class="fa-solid fa-code text-sm mr-3 text-brand-500"
          />
          Edit code
        </div>
      </a>
      <a
        class="transition duration-100 px-3 flex items-center h-8 text-sm font-medium rounded-sm outline-none data-[highlighted]:bg-brand-50 data-[highlighted]:text-brand-500 dark:data-[highlighted]:bg-neutral-400 dark:text-neutral-100 cursor-pointer text-neutral-400 hover:bg-brand-50 hover:text-brand-500 dark:hover:bg-neutral-550 gap-0"
        data-orientation="vertical"
        data-radix-collection-item=""
        href="/organization/123/audit-logs/general?targetId=0&targetType=HELM&projectId=123&environmentId=3840068565336064"
        role="menuitem"
        tabindex="-1"
      >
        <i
          aria-hidden="true"
          class="fa-solid fa-clock-rotate-left text-sm mr-3 text-brand-500"
        />
        See audit logs
      </a>
      <div
        class="px-3 flex items-center h-8 text-sm font-medium rounded-sm outline-none data-[highlighted]:bg-brand-50 data-[highlighted]:text-brand-500 dark:data-[highlighted]:bg-neutral-400 dark:text-neutral-100 cursor-pointer text-neutral-400 hover:bg-brand-50 hover:text-brand-500 dark:hover:bg-neutral-550"
        data-orientation="vertical"
        data-radix-collection-item=""
        role="menuitem"
        tabindex="-1"
      >
        <i
          aria-hidden="true"
          class="fa-solid fa-copy text-sm mr-3 text-brand-500"
        />
        Copy identifiers
      </div>
      <a
        class="transition duration-100 px-3 flex items-center h-8 text-sm font-medium rounded-sm outline-none data-[highlighted]:bg-brand-50 data-[highlighted]:text-brand-500 dark:data-[highlighted]:bg-neutral-400 dark:text-neutral-100 cursor-pointer text-neutral-400 hover:bg-brand-50 hover:text-brand-500 dark:hover:bg-neutral-550 gap-0"
        data-orientation="vertical"
        data-radix-collection-item=""
        href="/organization/123/project/123/environment/3840068565336064/application/0/settings/general"
        role="menuitem"
        tabindex="-1"
      >
        <i
          aria-hidden="true"
          class="fa-solid fa-gear text-sm mr-3 text-brand-500"
        />
        Open settings
      </a>
      <div
        class="px-3 flex items-center h-8 text-sm font-medium rounded-sm outline-none data-[highlighted]:bg-brand-50 data-[highlighted]:text-brand-500 dark:data-[highlighted]:bg-neutral-400 dark:text-neutral-100 cursor-pointer text-neutral-400 hover:bg-brand-50 hover:text-brand-500 dark:hover:bg-neutral-550"
        data-orientation="vertical"
        data-radix-collection-item=""
        role="menuitem"
        tabindex="-1"
      >
        <i
          aria-hidden="true"
          class="fa-solid fa-copy text-sm mr-3 text-brand-500"
        />
        Clone
      </div>
      <div
        aria-orientation="horizontal"
        class="my-1 h-[1px] w-full bg-neutral-200 dark:bg-neutral-600"
        role="separator"
      />
      <div
        class="px-3 flex items-center h-8 text-sm font-medium rounded-sm outline-none data-[highlighted]:bg-red-50 data-[highlighted]:text-red-600 cursor-pointer hover:bg-red-50 text-red-600"
        data-orientation="vertical"
        data-radix-collection-item=""
        role="menuitem"
        tabindex="-1"
      >
        <i
          aria-hidden="true"
          class="fa-solid fa-trash text-sm mr-3 text-red-600"
        />
        Delete service
      </div>
    </div>
  </div>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;
